xonsh/xoreutils/_which.py

```diff
@@ -191,21 +192,21 @@ def whichgen(command, path=None, verbose=0, exts=None):
     # Windows has the concept of a list of extensions (PATHEXT env var).
     if sys.platform.startswith("win"):
         if exts is None:
-            exts = os.environ.get("PATHEXT", "").split(os.pathsep)
+            exts = builtins.__xonsh_env__['PATHEXT']
```

Comment:
I think we should keep `_which.py` oblivious to xonsh, and just use it as a module. Then we still have the option to move it to an independent module. Before we just passed in `PathExts` through the `ext` argument to `_whichgen`